enhancements/update/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,197 @@
+
+
+### Risks and Mitigations
+
+### Container image content integrity
+The content of the cincinnati-graph-data repository may be altered by the process of being packaged by CPaaS and distributed via the container image registry.
+
+#### Mitigation
+A mitigation strategy is to sign the resulting image in the CPaaS pipeline, publish the signatures, and have Cincinnati verify the image after download.
+This is described throughout the [implementation section](#implementation-detailsnotesconstraints).
+
+### cincinnati-graph-data source repository integrity
+Not strictly related to this enhancement, is the fact that we are not signing the commits in the source repository in the first place.
+By this we effectively trust GitHub not to change the content.
+
+#### Mitigation
+OpenShift architects are aware of this risk and currently do not consider it sufficiently serious to want mitigation.
+If we decide to mitigate in the future, having build tooling require commits to be signed by a hard-coded list of authorized maintainers would be one possible approach.
+
+
+## Design Details
+
+### Test Plan
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+**Note:** *Section not required until targeted at a release.*
+
+- Maturity levels - `Developer Preview`, `Red Hat Staging`, `Red Hat Production`, `GA`
+
+##### `Developer Preview`
+- New Cincinnati plugin
+- Container image registry managed by and dedicated to prow CI
+- Integration tests using an image (to-be-determined: verifying against mocked signatures)
+- CPaaS pipeline is setup up to automatically build and publish a container image for changes to the master branch of the graph-data repository
+
+##### `Developer Preview` -> `Red Hat Staging`
+- Update the Cincinnati deployment template to use the new plugin.
+
+##### Red Hat Staging -> Red Hat Production
+- New plugin is successfully deployed to production
+- Adjust the Cincinnati Operator to use the new plugin
+
+##### Red Hat Production -> GA
+- Document the mirroring process for end-users
+- Definitive positive feedback from the ACM team and their use-case
+
+##### Removing a deprecated feature
+*Not applicable*
+
+### Upgrade / Downgrade Strategy
+*Not applicable*
+
+### Version Skew Strategy
+*Not applicable*
```

Comment:
Probably talk about Cincinnati vs. image/signature formats that are newer/older than what it understands here.  By linking [here][1]?  If Cincinnati hits a version it doesn't understand, how would it inform its operator?  Via a metric we can alert on?

[1]: https://github.com/openshift/cincinnati-graph-data/tree/6869639daa1d7de3eb2941250ed3dedc24ac55e2#schema-version

Reply:
I don't think this is in scope, as we're merely talking about changing the transport mechanism of the data here, not about the data itself.